test/util/utils.js

```diff
@@ -284,8 +289,134 @@ function objectToArray(input) {
   return result;
 }
 
+async function setupTestEnvironment(
+  contracts,
+  { returnClient = false, returnAccessController = false, bosonTokenAddress, forwarderAddress } = {}
```

Comment:
Properties `returnAccessController` and `returnClient` from `setupTestEnvironment` args seem unnecessary to me. Why don't you return the entire extraReturnValues by default and leave the decision of which properties to extract from `extraReturnValues` directly in the test files? Since there are no "heavy" behaviors inside the if statements, such as promises, it seems unnecessary to have these variables.